### PR TITLE
pm2 DA508/django restart DX

### PR DESCRIPTION
**Pull request by riatzukiza**
## Summary of Changes

This PR has tools I made for my self while developing on da508 integration that I expanded on a little bit for the backend. With this, we can work on the backend without restarting it and the changes are reflected instantly in development. 

Work is still being done on making sure it isn't breaking CI, otherwise you can use these files as they are to work locally.

Some of these may not be nessisary any more, I've included all the work I was doing at various staging in thinking about this. As I initially thought we had to do a lot more.
## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. `tdrs-rebuild-backend && tdrs-start-backend && tdrs-compose-backend logs -f`
2. wait for it to gurgle a little bit
3. When the logs have stopped, try making a change in either da508 or tdpservice

## Tabletop

- [ ] Ready for Tabletop (Reviewer check this)
- [ ] Tabletop scheduled (Assignee check this)
- [ ] Tabletop approval 

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [ ] [**_insert ACs here_**]
+ [ ] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

**Comment by riatzukiza** [comment]
There is no ticket for this one, this is based on work I had to do any way in order to test any of the da508 work. I keep having issues with it where the mysite example doesn't match up with tdp's admin panel. So here I am trying to codeify the steps I have to take to test da508 locally with a local tdp. Having to upload it to test pypi adds a lot of time to the dev loop. I am hoping here I can get a solution that basicly feels like real time you make changes to a local DA508, and you can look at a running instance of tdrs set up correctly and the changes are reflected.

File: tdrs-backend/Dockerfile [hunk 1]
```diff
@@ -15,17 +15,18 @@ RUN apt-get -y update
...
 WORKDIR /tdpapp/
 
 RUN groupadd -g ${gid} ${group} && useradd -u ${uid} -g ${group} -s /bin/sh ${user}
 
-RUN chown -R tdpuser /tdpapp && chmod u+x gunicorn_start.sh wait_for_services.sh
-
+RUN chown -R tdpuser /tdpapp && chmod u+x gunicorn_start.sh wait_for_services.sh pm2_start.sh install_nvm.sh
 
-CMD ["./gunicorn_start.sh"]
+CMD ["bash", "./install_nvm.sh"]
```
Comment (riatzukiza) [inline tdrs-backend/Dockerfile:32]: bad name, part of a thought process

File: tdrs-backend/Pipfile [hunk 2]
```diff
@@ -47,6 +46,7 @@ pyjwt = "==2.1.0"
 requests = "==2.26.0"
 wait-for-it = "==2.2.0"
 requests-mock = "==1.9.3"
+django-admin-508 = {editable = true, path = "./django-admin-508"}
```
[the remark this answers is not in the document]
Comment (riatzukiza) [reply tdrs-backend/Pipfile:49]: @raftmsohani This is the main issue I was talking about I wanted help on with this one.

[the remark this answers is not in the document]
Comment (raftmsohani) [reply tdrs-backend/Pipfile:49]: to be able to develop a package and testing it with Django app, we can forward the requirement to git branch. There are different ways to do it. See this: https://stackoverflow.com/questions/16584552/how-to-state-in-requirements-txt-a-direct-github-source
Then you can refer it to git branch temporarily and then revert it back to the original version after dev is finished

**Pull request closed without merging**